server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -156,6 +162,24 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             }
         }
 
+        @Override
+        public Query wildcardQuery(String value, MultiTermQuery.RewriteMethod method, QueryShardContext context) {
+            Query termQuery = termQuery(value, context);
+            if (termQuery instanceof MatchNoDocsQuery || termQuery instanceof MatchAllDocsQuery) {
+                return termQuery;
+            }
+
```

Comment:
We don't want to create a wildcard query for the `_type` field, this is not needed.
It should be possible to extend the new `ConstantFieldType` now that `_type` can have a single value (or none) ? 
Currently `TypeFieldType#termQuery` can return `Queries.newNonNestedFilter()` and that's not handled in the code below. Although, I don't think this is needed since we [exclude nested docs](https://github.com/elastic/elasticsearch/blob/b2ea32959103b93c03cc487ce475adde792f21e3/server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java#L268) for the entire query at the end of the query parsing so it should be safe to remove. IMO the `_type` field should solely return `MatchNoDocsQuery` and `MatchAllDocsQuery`, that's exactly what the `ConstantFieldType` provides :). 
@romseygeek what do you think ?

Reply:
> It should be possible to extend the new ConstantFieldType now that _type can have a single value (or none) ?

+1